uikit/viewmodels/src/main/java/com/sygic/maps/uikit/viewmodels/navigation/preview/RoutePreviewControlsViewModel.kt

```diff
@@ -0,0 +1,83 @@
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+
+package com.sygic.maps.uikit.viewmodels.navigation.preview
+
+import androidx.lifecycle.*
+import com.sygic.maps.tools.annotations.AutoFactory
+import com.sygic.maps.uikit.viewmodels.common.navigation.preview.RouteDemonstrationManager
+import com.sygic.maps.uikit.viewmodels.common.navigation.preview.state.DemonstrationState
+import com.sygic.maps.uikit.views.navigation.preview.RoutePreviewControls
+import com.sygic.maps.uikit.views.navigation.preview.state.PlayPauseButtonState
+
+/**
+ * A [RoutePreviewControlsViewModel] is a basic ViewModel implementation for the [RoutePreviewControls] class. It listens
+ * to the [RouteDemonstrationManager.demonstrationState] and set appropriate state to the [RoutePreviewControls] view.
+ * It also listens to the [RoutePreviewControls] user interaction callbacks and control the [RouteDemonstrationManager].
+ */
+@AutoFactory
+@Suppress("unused", "MemberVisibilityCanBePrivate")
+open class RoutePreviewControlsViewModel internal constructor(
+    private val routeDemonstrationManager: RouteDemonstrationManager
+) : ViewModel(), RoutePreviewControls.OnPlayPauseStateChangedListener {
+
+    val playPauseButtonState: MutableLiveData<PlayPauseButtonState> = MutableLiveData(PlayPauseButtonState.PLAY)
+
+    private val demonstrationStateObserver = Observer<DemonstrationState> { state ->
+        playPauseButtonState.value =
+            if (state == DemonstrationState.ACTIVE) PlayPauseButtonState.PAUSE else PlayPauseButtonState.PLAY
+    }
+
+    init {
+        routeDemonstrationManager.demonstrationState.observeForever(demonstrationStateObserver)
+    }
+
+    override fun onPlayPauseButtonStateChanged(state: PlayPauseButtonState) {
+        when (state) {
+            PlayPauseButtonState.PLAY -> routeDemonstrationManager.pause()
+            PlayPauseButtonState.PAUSE -> {
+                if (routeDemonstrationManager.demonstrationState.value == DemonstrationState.PAUSED) {
+                    routeDemonstrationManager.unPause()
+                } else {
+                    routeDemonstrationManager.restart()
```

Comment:
toto znamena, ze ked pocas behu stlacis drhukrat play, tak sa restartne, jo?

Reply:
druhýkrát play stisknout nelze (střídá se play / pause akce v jednom tlačítku), ale ještě to upravím, přijde mi to také s odstupem času trochu nelogické :)